packages/expo-linking/android/src/main/java/expo/modules/linking/ExpoLinkingModule.kt

```diff
@@ -0,0 +1,37 @@
+package expo.modules.linking
+
+import android.net.Uri
+import androidx.core.os.bundleOf
+import expo.modules.kotlin.modules.Module
+import expo.modules.kotlin.modules.ModuleDefinition
+
+class ExpoLinkingModule : Module() {
+  companion object {
+    var initialURL: Uri? = null
+    var onURLReceivedObservers: MutableSet<((Uri?) -> Unit)> = mutableSetOf()
+  }
+
+  private var onURLReceivedObserver: ((Uri?) -> Unit)? = null
+
+  override fun definition() = ModuleDefinition {
+    Name("ExpoLinking")
+
+    Events("onURLReceived")
+
+    Function("getLinkingURL") {
+      initialURL?.toString()
+    }
+
+    OnStartObserving("onURLReceived") {
+      val observer = { uri: Uri? ->
+        this@ExpoLinkingModule.sendEvent("onURLReceived", bundleOf("url" to uri?.toString()))
+      }
```

Comment:
```suggestion
			val weakModule = WeakReference(this@ExpoLinkingModule)
      val observer = { uri: Uri? ->
        weakModule.get()?.sendEvent("onURLReceived", bundleOf("url" to uri?.toString()))
      }
```
Just to be extra safe and not capture the module reference